tests/returntypes/test_torch_qnode_new.py

```diff
@@ -0,0 +1,2138 @@
+
+        jac = jacobian(circuit, a)
+
+        assert isinstance(jac, tuple)
+        assert len(jac) == 2
+
+        assert isinstance(jac[0], torch.Tensor)
+        assert jac[0].shape == ()
+
+        assert isinstance(jac[1], torch.Tensor)
+        assert jac[1].shape == (4,)
+
+    def test_jacobian_multiple_measurement_multiple_param(self, dev_name, diff_method, mode, shots):
+        """The jacobian of multiple measurements with a multiple params return a tuple of arrays."""
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because of probabilities.")
+        if shots is not None and diff_method in ("backprop", "adjoint"):
+            pytest.skip("Test does not support finite shots and adjoint/backprop")
+
+        dev = qml.device(dev_name, wires=2, shots=shots)
+
+        @qnode(dev, interface="torch", diff_method=diff_method, mode=mode)
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=0)
+            return qml.expval(qml.PauliZ(0)), qml.probs(wires=[0, 1])
+
+        a = torch.tensor(0.1, requires_grad=True)
+        b = torch.tensor(0.2, requires_grad=True)
+
+        jac = jacobian(circuit, (a, b))
+
+        assert isinstance(jac, tuple)
+        assert len(jac) == 2
+
+        assert isinstance(jac[0], tuple)
+        assert len(jac[0]) == 2
+        assert isinstance(jac[0][0], torch.Tensor)
+        assert jac[0][0].shape == ()
+        assert isinstance(jac[0][1], torch.Tensor)
+        assert jac[0][1].shape == ()
+
+        assert isinstance(jac[1], tuple)
+        assert len(jac[1]) == 2
+        assert isinstance(jac[1][0], torch.Tensor)
+        assert jac[1][0].shape == (4,)
+        assert isinstance(jac[1][1], torch.Tensor)
+        assert jac[1][1].shape == (4,)
+
+    def test_jacobian_multiple_measurement_multiple_param_array(
+        self, dev_name, diff_method, mode, shots
+    ):
+        """The jacobian of multiple measurements with a multiple params array return a single array."""
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because of probabilities.")
+        if shots is not None and diff_method in ("backprop", "adjoint"):
+            pytest.skip("Test does not support finite shots and adjoint/backprop")
+
+        dev = qml.device(dev_name, wires=2, shots=shots)
+
+        @qnode(dev, interface="torch", diff_method=diff_method, mode=mode)
+        def circuit(a):
+            qml.RY(a[0], wires=0)
+            qml.RX(a[1], wires=0)
+            return qml.expval(qml.PauliZ(0)), qml.probs(wires=[0, 1])
+
+        a = torch.tensor([0.1, 0.2], requires_grad=True)
+
+        jac = jacobian(circuit, a)
+
+        assert isinstance(jac, tuple)
+        assert len(jac) == 2  # measurements
+
+        assert isinstance(jac[0], torch.Tensor)
+        assert jac[0].shape == (2,)
+
+        assert isinstance(jac[1], torch.Tensor)
+        assert jac[1].shape == (4, 2)
```

Comment:
I think with JAX, `jac[0]` and `jac[1]` would also be tuples. Just double-checking: the reason why that's not the case here is that Torch allows only a single lvl of tuple nesting, correct? What problem does the pyteeifying solve?

Reply:
No here we have only one parameters and the results are a tuple (2, 4), therefore the jacobian is a single tuple (1 parameters with 2 measurements).

Here is the check in the Jax test:
```
        assert isinstance(jac, tuple)
        assert len(jac) == 2  # measurements

        assert isinstance(jac[0], jax.numpy.ndarray)
        assert jac[0].shape == (2,)

        assert isinstance(jac[1], jax.numpy.ndarray)
        assert jac[1].shape == (4, 2)
```

Reply:
My bad! Thanks.